File: gaia-sdk-javascript/src/graphql/request/type/CreatedIdentityImpulse.ts

import {Identity} from "./Identity";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";

/**
 * Impulse which indicates the result of a create identity impulse
 */
export class CreatedIdentityImpulse extends Array<(_:VariableRegistry) => string> {

    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the identity instance
     */
    public data = (config: (_:Identity) => void) => this.push((registry) => {
        const entity = new Identity();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/CreatedIdentityImpulse.py

from gaia_sdk.graphql.response.type.Identity import Identity

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class CreatedIdentityImpulse:
    """
    Impulse which indicates the result of a create identity impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the identity instance
    """
    @property
    def data(self) -> Identity:
        return Identity(self.dictionary.get("data"))
